backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/UsersController.java

```diff
@@ -25,26 +35,75 @@
-			@NotBlank(message = "id must not be null or blank")
-			@Parameter(description = "The id of the user.", example = "000000000", required = true)
-			@PathVariable String id) {
-		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(id));
-		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(id)).withSelfRel();
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId) {
+		final var subscriptions = subscriptionModelAssembler.toCollectionModel(subscriptionService.getSubscriptionsByUserId(userId));
+		final var selfLink = linkTo(methodOn(getClass()).getSubscriptionsByUserId(userId)).withSelfRel();
 		return subscriptionModelAssembler.wrapCollection(subscriptions, SubscriptionModel.class).add(selfLink);
+	}	
+
+
+	@PostMapping({ "/{userId}/subscriptions" })
+	@Operation(summary = "Create a subscription for a user.", operationId = "user-subscriptions-create")
+	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
+	public void create(
+			@RequestBody(required = true)
+			SubscriptionModel createSubscriptionRequest,
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId) throws Exception {
+		final Optional<AlertType> convertedAlertType = Optional.ofNullable(alertTypeService.readByCode(createSubscriptionRequest.getAlertType())).orElseThrow(() -> new Exception("alertType not found"));
+		createSubscriptionRequest.setAlertType(convertedAlertType.get().getId());
+		final var subscription = mapper.toDomain(createSubscriptionRequest);
+		log.debug("Creating subscription: {}", subscription);
+		subscriptionService.create(subscription);
 	}
 
+	@PostMapping({ "/{userId}/subscriptions/{subscriptionId}" })
+	@Operation(summary = "Update a subscription for a user.", operationId = "user-subscriptions-update")
+	@ApiResponse(responseCode = "204", description = "The request has been successfully processed.")
+	public void update(
+			@RequestBody(required = true)
+			SubscriptionModel updateSubscriptionRequest,
+			@NotBlank(message = "userId must not be null or blank")
+			@Parameter(description = "The user id of the user.", example = "000000000", required = true)
+			@PathVariable String userId,
+			@NotBlank(message = "subscriptionId must not be null or blank")
+			@Parameter(description = "The individaul subscription id. ", required = true)
+			@PathVariable String subscriptionId) throws Exception {
+		final Optional<AlertType> convertedAlertType = Optional.ofNullable(alertTypeService.readByCode(updateSubscriptionRequest.getAlertType())).orElseThrow(() -> new Exception("alertType not found"));
+		updateSubscriptionRequest.setAlertType(convertedAlertType.get().getId());
```

Comment:
See other comment about setting this value via the mapper.

Reply:
fixed